examples/wasm_custom_event.rs

```diff
@@ -0,0 +1,130 @@
+/// This example will show how to call a Rust function from JavaScript
+/// You should see the number 42 appear in browser console output when you click the button
+
+#[cfg(not(wasm_platform))]
+pub fn main() {
+    panic!("This example is only meant to be compiled for wasm target")
+}
+
+#[cfg(wasm_platform)]
+pub fn main() {
+    panic!("Please run `cargo run-wasm --example wasm_custom_event`")
+}
+
+#[cfg(wasm_platform)]
+mod wasm {
+    use std::cell::RefCell;
+
+    use wasm_bindgen::prelude::*;
+    use wasm_bindgen::JsCast;
+    use web_sys::HtmlScriptElement;
+    use winit::event::Event;
+    use winit::event::WindowEvent;
+    use winit::event_loop::EventLoop;
+    use winit::event_loop::EventLoopBuilder;
+    use winit::event_loop::EventLoopProxy;
+    use winit::window::Window;
+    use winit::window::WindowBuilder;
+
+    // Because EventLoopProxy is not Send, we need to wrap it in a RefCell and use thread_local!
+    thread_local! {
+        pub static EVENT_LOOP_PROXY: RefCell<Option<EventLoopProxy<CustomEvent>>> = RefCell::new(None);
+    }
+
+    // Function to be called from JS
+    fn wasm_call() -> u32 {
+        42
+    }
+
+    #[derive(Debug, Clone, Copy)]
+    pub enum CustomEvent {
+        WasmCall,
+    }
+
+    #[wasm_bindgen(start)]
+    pub fn run() {
+        console_log::init_with_level(log::Level::Debug).expect("error initializing logger");
+
+        let event_loop: EventLoop<CustomEvent> =
+            EventLoopBuilder::<CustomEvent>::with_user_event().build();
+
+        let event_loop_proxy = event_loop.create_proxy();
+
+        // Initialize the thread_local EVENT_LOOP_PROXY value
+        EVENT_LOOP_PROXY.with(move |proxy| {
+            proxy.replace(Some(event_loop_proxy));
+        });
+
+        let window = WindowBuilder::new().build(&event_loop).unwrap();
+
+        insert_canvas(&window);
+
+        event_loop.run(move |event, _, control_flow| {
+            control_flow.set_wait();
+
+            match event {
+                Event::WindowEvent {
+                    event: WindowEvent::CloseRequested,
+                    window_id,
+                } if window_id == window.id() => control_flow.set_exit(),
```

Comment:
We can skip that here, it's not a thing in Wasm.

Reply:
Skip this match arm?

```rust
Event::WindowEvent {
    event: WindowEvent::CloseRequested,
    window_id,
} if window_id == window.id() => control_flow.set_exit(),
```